src/Altinn.Platform/Altinn.Platform.Events/Tests/Mocks/PepWithPDPAuthorizationMockSI.cs

```diff
@@ -0,0 +1,380 @@
+        {
+            decisionRequest = await Enrich(decisionRequest);
+
+             XacmlPolicy policy = await GetPolicyAsync(decisionRequest);
+
+            PolicyDecisionPoint pdp = new PolicyDecisionPoint();
+            XacmlContextResponse xacmlContextResponse = pdp.Authorize(decisionRequest, policy);
+
+            return xacmlContextResponse;
+        }
+
+        private string GetInstanceID(XacmlJsonRequestRoot xacmlJsonRequest)
+        {
+            string instanceId = string.Empty;
+            foreach (XacmlJsonCategory category in xacmlJsonRequest.Request.Resource)
+            {
+                foreach (var atr in category.Attribute)
+                {
+                    if (atr.AttributeId.Equals(AltinnXacmlUrns.InstanceId))
+                    {
+                        instanceId = atr.Value;
+                        break;
+                    }
+                }
+            }
+
+            return instanceId;
+        }
+
+        public async Task<bool> GetDecisionForUnvalidateRequest(XacmlJsonRequestRoot xacmlJsonRequest, ClaimsPrincipal user)
+        {
+            XacmlJsonResponse response = await GetDecisionForRequest(xacmlJsonRequest);
+            return DecisionHelper.ValidatePdpDecision(response.Response, user);
+        }
+
+        public async Task<XacmlContextRequest> Enrich(XacmlContextRequest request)
+        {
+            await EnrichResourceAttributes(request);
+
+            return request;
+        }
+
+        private async Task EnrichResourceAttributes(XacmlContextRequest request)
+        {
+            XacmlContextAttributes resourceContextAttributes = request.GetResourceAttributes();
+            XacmlResourceAttributes resourceAttributes = GetResourceAttributeValues(resourceContextAttributes);
+
+            bool resourceAttributeComplete = false;
+
+            if (!string.IsNullOrEmpty(resourceAttributes.OrgValue) &&
+                !string.IsNullOrEmpty(resourceAttributes.AppValue) &&
+                !string.IsNullOrEmpty(resourceAttributes.InstanceValue) &&
+                !string.IsNullOrEmpty(resourceAttributes.ResourcePartyValue) &&
+                !string.IsNullOrEmpty(resourceAttributes.TaskValue))
+            {
+                // The resource attributes are complete
+                resourceAttributeComplete = true;
+            }
+            else if (!string.IsNullOrEmpty(resourceAttributes.OrgValue) &&
+                !string.IsNullOrEmpty(resourceAttributes.AppValue) &&
+                string.IsNullOrEmpty(resourceAttributes.InstanceValue) &&
+                !string.IsNullOrEmpty(resourceAttributes.ResourcePartyValue) &&
+                string.IsNullOrEmpty(resourceAttributes.TaskValue))
+            {
+                // The resource attributes are complete
+                resourceAttributeComplete = true;
+            }
+            else if (!string.IsNullOrEmpty(resourceAttributes.OrgValue) &&
```

Comment:
Could we not simplify this mock? I don't see how we would need any cases other than the one below.

Reply:
simplified